velox/type/tests/DecimalTest.cpp

```diff
@@ -287,84 +291,202 @@ TEST(DecimalAggregateTest, adjustSumForOverflow) {
+  assertRescaleFloatingPoint<float, int128_t>(
+      -3333.03, DECIMAL(20, 1), -33'330);
+  assertRescaleFloatingPoint<float, int128_t>(
+      -3333.03,
+      DECIMAL(38, 18),
+      HugeInt::parse("-333303" + std::string(16, '0')));
+
+  assertRescaleFloatingPoint<float, int64_t>(
+      -2222.02, DECIMAL(10, 4), -22'220'200);
+  assertRescaleFloatingPoint<float, int128_t>(
+      -2222.02, DECIMAL(20, 1), -22'220);
+  assertRescaleFloatingPoint<float, int128_t>(
+      -2222.02,
+      DECIMAL(38, 18),
+      HugeInt::parse("-222202" + std::string(16, '0')));
+
+  assertRescaleFloatingPoint<float, int64_t>(-1.0, DECIMAL(10, 4), -10'000);
+  assertRescaleFloatingPoint<float, int128_t>(-1.0, DECIMAL(20, 1), -10);
+  assertRescaleFloatingPoint<float, int128_t>(
+      -1.0, DECIMAL(38, 18), -1'000'000'000'000'000'000);
+
+  assertRescaleFloatingPoint<float, int64_t>(0.00, DECIMAL(10, 4), 0);
+  assertRescaleFloatingPoint<float, int128_t>(0.00, DECIMAL(20, 1), 0);
+  assertRescaleFloatingPoint<float, int128_t>(0.00, DECIMAL(38, 18), 0);
+
+  assertRescaleFloatingPoint<float, int64_t>(100, DECIMAL(10, 4), 1'000'000);
+  assertRescaleFloatingPoint<float, int128_t>(100, DECIMAL(20, 1), 1'000);
+  assertRescaleFloatingPoint<float, int128_t>(
+      100, DECIMAL(38, 18), HugeInt::parse("100" + std::string(18, '0')));
+
+  assertRescaleFloatingPoint<float, int64_t>(
+      9999.99, DECIMAL(10, 4), 99'999'900);
+  assertRescaleFloatingPoint<float, int128_t>(9999.99, DECIMAL(20, 1), 100'000);
+  assertRescaleFloatingPoint<float, int128_t>(
+      9999.99,
+      DECIMAL(38, 18),
+      HugeInt::parse("999999" + std::string(16, '0')));
+
+  assertRescaleFloatingPoint<float, int128_t>(0.95, DECIMAL(3, 1), 10);
+  assertRescaleFloatingPoint<float, int128_t>(
+      10.03, DECIMAL(38, 18), HugeInt::parse("1003" + std::string(16, '0')));
+  assertRescaleFloatingPoint<float, int128_t>(
+      0.034567, DECIMAL(38, 18), 34'567'000'000'000'000);
+  assertRescaleFloatingPoint<float, int128_t>(
+      0.999999999999999, DECIMAL(38, 18), 1'000'000'000'000'000'000);
+  assertRescaleFloatingPoint<float, int128_t>(
+      0.123456, DECIMAL(38, 18), 123'456'000'000'000'000);
+  assertRescaleFloatingPoint<float, int64_t>(21.5455, DECIMAL(12, 3), 21546);
+
+  assertRescaleFloatingPointFail<float, int128_t>(
+      std::numeric_limits<float>::max(), DECIMAL(38, 38), "Result overflows.");
+  assertRescaleFloatingPoint<float, int128_t>(
+      std::numeric_limits<float>::min(), DECIMAL(38, 2), 0);
+  assertRescaleFloatingPointFail<float, int128_t>(
+      std::numeric_limits<float>::lowest(),
+      DECIMAL(38, 2),
+      "Result overflows.");
+
+  assertRescaleFloatingPointFail<float, int64_t>(
+      NAN, DECIMAL(10, 2), "The input value should be finite.");
+  assertRescaleFloatingPointFail<float, int64_t>(
+      INFINITY, DECIMAL(10, 2), "The input value should be finite.");
+
+  assertRescaleFloatingPointFail<float, int64_t>(
```

Comment:
Wondering if it helps to create an array of all these invalid values and check them in a loop.

```
for (auto v : invalidInputs) {
   assertRescaleRealFails(v, DECIMAL(10, 2), "Result overflows.");
}
```

Reply:
Updated as suggested. Thanks.